app/modules/SecurityModule.scala

```diff
@@ -56,6 +59,28 @@ class SecurityModule extends AbstractModule {
     val clients = new Clients(oidcClient)
     val config = new Config(clients)
     config.setHttpActionAdapter(new FrontendHttpActionAdaptor())
+    config.addAuthorizer("custom", new CustomAuthoriser)
     config
   }
 }
+
+/*
+pac4j has a default authoriser which checks for csrf tokens. The problem
+is that it doesn't work and there's not a lot of documentation on how to
+make it work.
+
+The other problem is that play has csrf checkers which are more than
+adequate so what we need is to turn off the pac4j one.
+
+If you don't provide an authoriser, it gives you the csrf one by default
+so the trick is to give it a dummy authoriser to stop it checking the
+csrf one. Which is what this does.
+
+This has no effect on whether logged out users can see or not see
+certain pages. This is authorisation which we're not doing in the front
+end, although if we do decide to do front end authorisation, we can
+expand the CustomAuthoriser class to something more useful.
+*/
+class CustomAuthoriser extends Authorizer[CommonProfile] {
```

Comment:
This might be a moot point if we add actual authorization in the Pac4J layer, but I think you can delete this authoriser if you change the secureAction in OidcSecurity from this:

```
Secure("OidcClient", authorizers = "custom")
```

to this:

```
Secure("OidcClient", authorizers = DefaultAuthorizers.NONE)
```

Reply:
Suzanne will be returning to this issue outwith this branch as it is a separate problem.

Reply:
Fixed here: https://github.com/nationalarchives/tdr-transfer-frontend/pull/52